cmd/adwatchd/integration_tests/adwatchd_service_test.go

```diff
@@ -194,3 +194,59 @@ Watched directories:
 	// Get actual status
 	require.Equal(t, want, getStatus(t, app), "Service status doesn't match")
 }
+
+func TestServiceConfigFlagUsage(t *testing.T) {
+	tests := map[string]struct {
+		wantConfig bool
+	}{
+		// Subcommands not allowing a config flag
+		"start":     {},
+		"restart":   {},
+		"uninstall": {},
+		"status":    {},
+
+		// Subcommands allowing a config flag
+		"install": {wantConfig: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			r, w, err := os.Pipe()
+			require.NoError(t, err, "Setup: pipe shouldn't fail")
+
+			app := commands.New()
+
+			changeAppArgs(t, app, "", "service", name, "-c", "badconf")
+
+			err = app.Run()
+			if tc.wantConfig {
+				require.ErrorContains(t, err, "invalid configuration file")
```

Comment:
Here, you probably want the test to continue in case of error to check for the second require below. I would use then assert for both (it’s basically 2 tests one after another, but I understand that you want to keep both in the same place)